core/src/main/java/org/infinispan/xsite/GlobalXSiteAdminOperation.java

```diff
@@ -0,0 +1,128 @@
+      return performOperation(operations -> operations.takeSiteOffline(site));
+   }
+
+   @ManagedOperation(
+         description = "Brings the given site back online on all the caches.",
+         displayName = "Brings the given site back online on all the caches."
+   )
+   public String bringSiteOnline(@Parameter(name = "site", description = "The name of the backup site") String site) {
+      return performOperation(operations -> operations.bringSiteOnline(site));
+   }
+
+   @ManagedOperation(
+         displayName = "Push state to site",
+         description = "Pushes the state of all caches to the corresponding remote site if the cache backups to it. " +
+               "The remote site will be bring back online",
+         name = "pushState"
+   )
+   public final String pushState(@Parameter(description = "The destination site name", name = "SiteName") String site) {
+      return performOperation(operations -> operations.pushState(site));
+   }
+
+   @ManagedOperation(
+         displayName = "Cancel Push State",
+         description = "Cancels the push state on all the caches to remote site.",
+         name = "CancelPushState"
+   )
+   public final String cancelPushState(@Parameter(description = "The destination site name", name = "SiteName")
+                                       final String site) {
+      return performOperation(operations -> operations.cancelPushState(site));
+   }
+
+   private String performOperation(Operation operation) {
+      final Iterator<PerCacheXSiteAdminOperation> iterator = collectXSiteAdminOperation().iterator();
+      if (!iterator.hasNext()) {
+         return XSiteAdminOperations.SUCCESS;
+      }
+      StringBuilder builder = new StringBuilder();
+      while (iterator.hasNext()) {
+         PerCacheXSiteAdminOperation xsiteAdminOperation = iterator.next();
+         try {
+            String result = operation.execute(xsiteAdminOperation.xSiteAdminOperations);
+            builder.append(xsiteAdminOperation.cacheName).append(": ").append(result);
+            if (iterator.hasNext()) {
+               builder.append(CACHE_DELIMITER);
+            }
+         } catch (Exception e) {
+            builder.append("Exception on ").append(xsiteAdminOperation.cacheName).append(": ").append(e.getMessage());
+         }
+      }
+      return builder.length() == 0 ? XSiteAdminOperations.SUCCESS : builder.toString();
+   }
+
+   private Collection<PerCacheXSiteAdminOperation> collectXSiteAdminOperation() {
+      Collection<String> cacheNames = cacheManager.getCacheNames();
+      List<PerCacheXSiteAdminOperation> operations = new ArrayList<>(cacheNames.size() + 1);
+      for (String cacheName : cacheNames) {
+         addIfBackupRemotely(cacheManager.getCache(cacheName, false), operations);
+      }
+      addIfBackupRemotely(cacheManager.getCache(), operations);
+      return operations;
+   }
+
+   private interface Operation {
+      String execute(XSiteAdminOperations operations);
```

Comment:
Executing an operation on an `operations` object is a bit too meta here, how about renaming the parameter (and the other local variables) to `mbeans`?